sql/rpl_binlog_sender.cc

```diff
@@ -1030,7 +1038,73 @@ int Binlog_sender::send_format_description_event(IO_CACHE *log_cache,
   if (event_checksum_on() && event_updated)
     calc_event_checksum(event_ptr, event_len);
 
-  DBUG_RETURN(send_packet());
+  if (m_event_checksum_alg != binary_log::BINLOG_CHECKSUM_ALG_UNDEF &&
+      m_event_checksum_alg != binary_log::BINLOG_CHECKSUM_ALG_OFF)
+    event_len-= BINLOG_CHECKSUM_LEN;
+
+  Format_description_log_event *new_fdle= NULL;
+
+  try 
+  {
+    new_fdle= new Format_description_log_event(reinterpret_cast<char*>(event_ptr), event_len, m_fdle.get());
+  }
+  catch(...)
+  {
+    set_fatal_error("Corrupt Format_description event found or out-of-memory");
```

Comment:
Corrupt event cannot result in exception thrown, right? So just nothrow new

Reply:
They should not, but actually they result in exceptions. That is why we needed checksums to be on for encryption. It can throw in situation when checksums are disabled, type code of the event is correct (Format description in this case), but the body of the event is different than assumed. I got a looot of exception when I was sending encrypted events to slave (from time to time the event type was correct, but the event body was garbage from slave pov).

Reply:
I do not have a specific list and it could be hard to get one. MySQL code is not using them, but you will easily get a server crash if you start sending garbage data to slave and checksum will be turned off.

Reply:
That is very suspicious - are you saying that some C++ exceptions are getting thrown from non-exceptions using code? Otherwise how does try/catch protect from crashes?